src/utils/xhr-utils.js

```diff
@@ -376,6 +383,11 @@ export function verifyAmpCORSHeaders(win, response, init) {
     userAssert(false, 'Response must contain the' +
         ` ${ALLOW_SOURCE_ORIGIN_HEADER} header`);
   }
+  devAssert(accessControlAllowOrigin,
```

Comment:
`devAssert`s will be striped from production. Also, if there were no accessControllAllowOrigin header on the response, we wouldn't have a response to interact with! Unless it opaque, maybe?

Reply:
> `devAssert`s will be striped from production

Yes, I think the goal is to catch this only in early development and the message/error is to signal to the developer that CORS testing needs to be done. Which seems to be something they tend to omit.

#21313 is also another effort to make the developer more aware about CORS testing.

Reply:
> to signal to the developer that CORS testing needs to be done

You're confusing `devAssert` (assert to the AMP Core developers) with `userAssert` (assert to the publisher's developers).

Reply:
ah gotcha..oops. Thanks